aws/data_source_aws_api_gateway_rest_api.go

```diff
@@ -0,0 +1,88 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsApiGatewayRestApi() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsApiGatewayRestApiRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"root_resource_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsApiGatewayRestApiRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	params := &apigateway.GetRestApisInput{}
+
+	target := d.Get("name")
+	var matchedApis []*apigateway.RestApi
+	log.Printf("[DEBUG] Reading API Gateway REST APIs: %s", params)
+	err := conn.GetRestApisPages(params, func(page *apigateway.GetRestApisOutput, lastPage bool) bool {
+		for _, api := range page.Items {
+			if *api.Name == target {
```

Comment:
To prevent a potential crash, we should wrap this with `aws.StringValue(api.Name)` in the unlikely scenario that its returned as `nil`